### PR TITLE
fix bug found when transmitter not at surface.

### PR DESCRIPTION
**Pull request by dccowan**
I found a bug in the forward simulation for em1dfm for a layered Earth. The bug happens when the transmitter is not on the surface. I have fixed the code and validated the new code against cyl mesh and octree codes in SimPEG.

image

### CODE REVIEW

File: geoana/em/fdem/layered.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def magnetic_field(self, xyz, field="secondary"):
 
         if np.any(xyz[:, 2] < 0.0):
             raise ValueError("Cannot compute fields below the surface")
-        h = self.location[0]
```
Comment (jcapriot) [inline geoana/em/fdem/layered.py:95]: Thanks for catching this bug, simple fix though!

File: geoana/em/fdem/layered.py [hunk 2]
```diff
@@ -121,7 +121,7 @@ def magnetic_field(self, xyz, field="secondary"):
         rTE = rTE.reshape((n_frequency, *lambd.shape))
 
         # secondary is height of receiver plus height of source
-        rTE *= np.exp(-lambd*(xyz[:, -1] + h)[:, None])
+        rTE *= np.exp(-lambd*(dxyz[:, -1] + 2*h)[:, None])
```
Comment (jcapriot) [inline geoana/em/fdem/layered.py:124]: These two lines are equivalent to each other (dxyz + 2*h) = ((xyz - location) + 2*location)=(xyz + location)

In my opinion, the original is clearer.

**Review by jcapriot — COMMENTED** [review]
Thanks for double checking

File: geoana/em/fdem/layered.py (hunk 2 above)
Comment (dccowan) [reply geoana/em/fdem/layered.py]: Sure. I'll change it back

**Review by jcapriot — APPROVED** [review]

**Pull request merged**